lib/src/back_end/solution.dart

```diff
@@ -137,10 +158,15 @@ class Solution implements Comparable<Solution> {
   /// piece and yields further solutions for each state that piece can have.
   List<Solution> expand(Piece root, int pageWidth) {
     if (_nextPieceToExpand case var piece?) {
-      return [
-        for (var state in piece.states)
-          Solution._(root, pageWidth, _state.cloneWith(piece, state))
-      ];
+      var result = <Solution>[];
+      for (var state in piece.states) {
+        var stateSet = _state.tryBind(piece, state);
```

Comment:
[out of scope for this PR] Can we consider renaming `_state` to `_stateSet` or similar? The distinction between `_state` and `state` here is too subtle IMO.

Reply:
Yeah, I agree that's much better. I'll just do it now.